gomilter.go

```diff
@@ -343,6 +343,19 @@ func Go_xxfi_close(ctx *C.SMFICTX) C.sfsistat {
 	if milter.GetDebug() {
 		logger.Printf("Close callback returned: %d\n", code)
 	}
+
+	// Verify if the private data from ctx is already NULL 
+	// Call libmilter smfi_getpriv to get a pointer to our data
+	CArray := (*byte)(C.smfi_getpriv(ctx))
+	pointerToData := unsafe.Pointer(CArray)
+	if pointerToData != nil {
+		lenStart := uintptr(unsafe.Pointer(CArray))
+		// Free the data malloc'ed by C
+		C.free(unsafe.Pointer(lenStart))
+		C.smfi_setpriv(ctx, nil)
+	}
```

Comment:
What's the difference to lines 460/461? Shouldn't we fix it there?

Reply:
C.free just frees the memory, and with smfi_setpriv we set ctx->ctx_privdata to nil 

The issue on 460/461 we rely on int2ctx to get the right pointer to our ctx struct. Which is not happening, this could also be due to the complex struct been encoded on ctx aka Envelope